### ISSUE TITLE
Logic / JS issues

### ISSUE THREAD

**Issue by ricardocasares** [ricardocasares/coding_task#3]
### Improvements
- [The `show` property of the interface](https://github.com/ricardocasares/coding_task/blob/main/src/components/ToDo.tsx#L8) seems to be redundant as we already have the `completed` prop that can be used for the same purpose.
- We've noticed this [part of code](https://github.com/ricardocasares/coding_task/blob/main/src/components/Navbar.tsx#L24) could be improved as it contains a lot of logic for a simple css `className` toggle. Hint, you may use state to check which filter is active.

### Logic issues
- If I set the navigation filter to `Pending`, I see a list of pending todos. If I then click the `Todo Status` the status should be changed to `completed` and now the todo is not visible inside the `Pending` view. If I switch the navigation filter to `Complete` I should find the previously completed todo on this list. As well I should find the todo item when the filter is set to `All`

**Comment by tkondek** [comment]
@Logic issues
Then I had to missunderstand this one "Given that Todos Filter Links is set to Pending, when clicking Todo Status of a todo it will remove that todo from the Todos Containe". 
I though that if list is in a pending filter then if pressed todo status the task should be completely removed from tasks.  Am I wrong ?

**Comment by ricardocasares** [comment]
@tkondek yes, I understand that the phrasing in the task description might have been confusing, but ultimately it should work this way:

### Filter set on `All`

Clicking on `Todo Status` toggles todo state between `Pending` and `Complete` status, without "removing" it from the list. By removing I'm talking about visibility on the `Todos Container`.

### Filter set on `Pending`

Clicking on `Todo Status` changes todo state from `Pending` to `Complete` status and the todo item disappears from the `Todos Container` because the filter is set to `Pending` only.

### Filter set on `Complete`

Clicking on `Todo Status` changes todo state from `Complete` to `Pending` status and the todo item disappears from the `Todos Container` because the filter is set to `Complete` only.

**Comment by ricardocasares** [comment]
@tkondek let me know if everything is clear

**Comment by tkondek** [comment]
Yes, that make more sens now. That was quite strange behaviour :). It
should be quick fix then.

W dniu pt., 22.01.2021 o 13:06 Ricardo Casares <notifications@github.com>
napisał(a):

**Comment by tkondek** [comment]
Hi Ricardo,
where can I find user name? There are only id, userId, title and completed properties on https://jsonplaceholder.typicode.com/todos. Do I miss something?

**Comment by ricardocasares** [comment]
Hi @tkondek you can find that in the users endpoint please read the json placeholder site and guidelines

https://jsonplaceholder.typicode.com/guide/
https://jsonplaceholder.typicode.com/users
